global-dispatcher/cmd/dispatch/base.go

```diff
@@ -47,9 +49,13 @@ func DispatchSessions(ctx context.Context, requestID string) (uint32, error) {
 		return 0, shared.ErrNoCacheClientProvided
 	}
 
-	db, err := database.ClientFromURI(ctx, mongoConnectionString, mongoDatabase)
+	dbClient, err := database.NewPHDClient(dbclient.Config{
+		BaseURL: phdBaseURL,
+		APIKey:  phdAPIKey,
```

Comment:
If I'm not mistakten, global services only reads from PHD, correct? If that's the case I'd recommend using the `NewReadOnlyDBClient` , which returns an interface that provides only the read methods for interacting with PHD.

I would also suggest setting the HTTP retries and timeout in the config struct.

Reply:
resolved in the `phd.go` file